### PR TITLE
Merge TechNexion meta-mender-community changes to upstream

### PR DESCRIPTION
**Pull request by po-cheng-technexion**
Merging forked TechNexion meta-mender-community
Minor changes to .xml filename and tn-mender-setup.sh
Remaining commits are related to mender-modules and recipes to build mender-artifacts

### CODE REVIEW

File: meta-mender-tn-imx-bsp/README.md [hunk 1]
```diff
@@ -23,8 +23,8 @@ that have Mender integrated.
 
 ```
 mkdir mender-tn-imx-bsp && cd mender-tn-imx-bsp
-repo init -u https://github.com/TechNexion/tn-imx-yocto-manifest.git -b zeus_5.4.y-next -m imx-5.4.24-2.1.0.xml
-wget --directory-prefix .repo/local_manifests https://raw.githubusercontent.com/mendersoftware/meta-mender-community/zeus/meta-mender-tn-imx-bsp/scripts/imx-5.4.24-2.1.0_demo_mender.xml
+repo init -u https://github.com/TechNexion/tn-imx-yocto-manifest.git -b zeus_5.4.y-next -m imx-5.4.47-2.2.0.xml
+wget --directory-prefix .repo/local_manifests https://raw.githubusercontent.com/mendersoftware/meta-mender-community/zeus/meta-mender-tn-imx-bsp/scripts/mender-technexion.xml
```
Comment (mirzak) [inline meta-mender-tn-imx-bsp/README.md:27]: This is ok.

Note that it is a user-facing breaking change, e.g we also need to update https://hub.mender.io/t/technexion-pico-pi-imx8m-mini/2689

**Review by mirzak — CHANGES_REQUESTED** [review]
Thank you for your contributions.

First of I would like to break this PR up in to several smaller ones as they independent changes that need to be discussed. 

1. First and last commit, there is not much to discuss, create a separate PR for these and that can be merged.

2. Addition of `directory`, `single-file`, `docker`, `dir-overlay`, `script` update modules

    I am not convinced if we should all of these. Do you have use-cases for all of these? The reason I am hesitant is I would like to avoid adding things if they are not used and would rather add things on-demand. Mostly trying to avoid a maintenance burden if things are not used.

    There is also a possibility to consolidate these change, e.g `mender-**-artifacts.bb` recipes share a lot of code.

3. package-mender-artifacts.bbclass 

    I find this one very interesting and would keep this in a separate PR as I would like to discuss this one in more detail.

File: meta-mender-tn-imx-bsp/README.md (hunk 1 above)
Comment (po-cheng-technexion) [reply meta-mender-tn-imx-bsp/README.md:27]: Thank you for the feedback, I'll close this pull request and PR in 3 bunches accordingly.

**Comment by po-cheng-technexion** [comment]
Force push those two straight-forward commits to our forked repository. Hope this makes it simpler

**Review by mirzak — APPROVED** [review]

**Pull request merged**